app/forms/webauthn_verification_form.rb

```diff
@@ -60,8 +63,9 @@ def valid_assertion_response?(protocol)
   end
 
   def public_key
-    WebauthnConfiguration.
-      where(user_id: user.id, credential_id: @credential_id).take.credential_public_key
+    @webauthn_configuration = WebauthnConfiguration.
+      where(user_id: user.id, credential_id: @credential_id).first
```

Comment:
Not sure if this would be the case, but should this form halt to notice a WebAuthConfiguration is missing?

```suggestion
    @webauthn_configuration = WebauthnConfiguration.
      find_by!(user: user, credential_id: @credential_id)
```

IMO, an `AR:NotFound` would be nicer than the almighty `NoMethodError (undefined method 'credential_public_key' for nil:NilClass)`, again, iff the system has this use case, otherwise please disregard my suggestion.

Reply:
This is a good reminder, I think we should leverage the AR associations for this instead of querying by ids/columns too as long as we're here:

```suggestion
    @webauthn_configuration = user.webauthn_configurations.find_by!(credential_id: @credential_id)
```